src/main/kotlin/org/jitsi/jicofo/jibri/JibriRecorder.kt

```diff
@@ -0,0 +1,198 @@
+/*
+ * Jicofo, the Jitsi Conference Focus.
+ *
+ * Copyright @ 2015-Present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.jicofo.jibri
+
+import org.apache.commons.lang3.StringUtils
+import org.jitsi.impl.protocol.xmpp.XmppProvider
+import org.jitsi.jicofo.JitsiMeetConferenceImpl
+import org.jitsi.jicofo.jibri.JibriConfig.Companion.config
+import org.jitsi.jicofo.jibri.JibriSession.StartException
+import org.jitsi.jicofo.jibri.JibriSession.StartException.AllBusy
+import org.jitsi.jicofo.jibri.JibriSession.StartException.NotAvailable
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.xmpp.extensions.jibri.JibriIq
+import org.jitsi.xmpp.extensions.jibri.JibriIq.RecordingMode
+import org.jitsi.xmpp.extensions.jibri.JibriIq.Status
+import org.jitsi.xmpp.extensions.jibri.RecordingStatus
+import org.jivesoftware.smack.packet.IQ
+import org.jivesoftware.smack.packet.XMPPError
+import org.jitsi.jicofo.util.ErrorResponse.create as error
+
+/**
+ * Handles conference recording through Jibri.
+ * Waits for updates from [JibriDetector] about recorder instance
+ * availability and publishes that information in Jicofo's MUC presence.
+ * Handles incoming Jibri IQs coming from conference moderator to
+ * start/stop the recording.
+ *
+ * @author Pawel Domas
+ * @author Sam Whited
+ */
+class JibriRecorder(
+    conference: JitsiMeetConferenceImpl,
+    xmppProvider: XmppProvider,
+    jibriDetector: JibriDetector,
+    parentLogger: Logger
+) : BaseJibri(
+    conference,
+    xmppProvider,
+    parentLogger,
+    jibriDetector
+) {
+    /**
+     * The current recording session or <tt>null</tt>.
+     */
+    private var jibriSession: JibriSession? = null
+
+    /**
+     * {@inheritDoc}
+     */
+    override fun dispose() {
+        jibriSession?.stop(null)
+        super.dispose()
```

Comment:
The Java code nulled `jibriSession` here, presumably to prevent stopping the session twice - why did you remove it?

Reply:
I brought it back 👍